src/adam_core/propagator/adam_assist.py

```diff
@@ -517,10 +523,127 @@ def _detect_impacts(
                 variant_id=[],
             )
         return results, earth_impacts
-
+    
+    def _add_light_time(
+        self, 
+        orbits,
+        observers,
+        lt_tol: float = 1e-10,
+    ):  
+        orbits_aberrated = orbits.empty()
+        lts = np.zeros(len(orbits))
+        for i, (orbit, observer) in enumerate(zip(orbits, observers)):
+            lt0 = 0
+            dlt = 1
+            while dlt > lt_tol:
+                observer_position = observer.coordinates.values[0,:3]
+                orbit_i = orbit
```

Comment:
What's the purpose of renaming here, since these will be the same object in memory? I was confused because we concat the `orbit` instance at the end but we are modifying `orbit_i` and I had to realize they were the same.